### PR TITLE
Update Self Tests

### PR DESCRIPTION
**Pull request by MrJnrman**
This PR adds an api self test to ensure that the `new_product_editor` feature can be enabled correctly. It also updates the existing snapshots based on the changes in https://github.com/Automattic/compatibility-dashboard/pull/414

### CODE REVIEW

File: _tests/tests/__snapshots__/ApiTest__test_api_valid_features__1.php
Comment (Luc45) [inline _tests/tests/__snapshots__/ApiTest__test_api_valid_features__1.php]: Awesome, we can see that it's activated here. However, there are some other notices being thrown around, I wonder if it has something to do with the way we enabled the feature plugin in the SUT?

```json
                   "count": "4287",
                    "message": "PHP Warning: array_filter() expects parameter 1 to be array, null given in \\/var\\/www\\/html\\/wp-content\\/plugins\\/woocommerce\\/src\\/Internal\\/Admin\\/FeaturePlugin.php on line 211"
                },
                {
                    "count": "4287",
                    "message": "PHP Warning: array_keys() expects parameter 1 to be array, null given in \\/var\\/www\\/html\\/wp-content\\/plugins\\/woocommerce\\/src\\/Internal\\/Admin\\/FeaturePlugin.php on line 211"
                },
                {
                    "count": "3676",
                    "message": "PHP Warning: array_diff(): Expected parameter 1 to be an array, null given in \\/var\\/www\\/html\\/wp-content\\/plugins\\/woocommerce\\/src\\/Admin\\/Features\\/Features.php on line 184"
                },
                {
                    "count": "3676",
                    "message": "PHP Warning: array_values() expects parameter 1 to be array, null given in \\/var\\/www\\/html\\/wp-content\\/plugins\\/woocommerce\\/src\\/Admin\\/Features\\/Features.php on line 184"
                },
                {
                    "count": "3676",
                    "message": "PHP Warning: in_array() expects parameter 2 to be array, null given in \\/var\\/www\\/html\\/wp-content\\/plugins\\/woocommerce\\/src\\/Admin\\/Features\\/Features.php on line 195"
                },
                {
                    "count": "611",
                    "message": "PHP Warning: Invalid argument supplied for foreach() in \\/var\\/www\\/html\\/wp-content\\/plugins\\/woocommerce\\/src\\/Admin\\/Features\\/Features.php on line 135"
                },
```

File: _tests/tests/__snapshots__/E2eTest__test_e2e_no_op__1.php
Comment (Luc45) [inline _tests/tests/__snapshots__/E2eTest__test_e2e_no_op__1.php]: Any idea why the Product editor tests started to fail on E2E, even though we didn't tweaked the e2e test env settings?

File: _tests/api/valid_features/woocommerce-product-feeds/woocommerce-product-feeds.php
Comment (Luc45) [inline _tests/api/valid_features/woocommerce-product-feeds/woocommerce-product-feeds.php]: This should be `add_filter`, returning `$features` at the end. That's why we are receiving the warnings on the test log

Comment (MrJnrman) [reply _tests/api/valid_features/woocommerce-product-feeds/woocommerce-product-feeds.php]: Good catch!

File: _tests/tests/__snapshots__/E2eTest__test_e2e_no_op__1.php
Comment (MrJnrman) [reply _tests/tests/__snapshots__/E2eTest__test_e2e_no_op__1.php]: I think this was from a previous run when I was modifying the e2e tests. Let me update the snapshots and see if it persists.

Comment (MrJnrman) [reply _tests/tests/__snapshots__/E2eTest__test_e2e_no_op__1.php]: Turns out that wasn't the reason. I checked the [test itself](https://github.com/Automattic/compatibility-dashboard/blob/trunk/ci/e2e/tests/new-product-editor/new-product-editor.spec.js) and it looks like it should fail if that specific test isn't skipped.

I think we should skip the product editor tests in general

File: _tests/tests/__snapshots__/ApiTest__test_api_valid_features__1.php
Comment (MrJnrman) [reply _tests/tests/__snapshots__/ApiTest__test_api_valid_features__1.php]: Turns out it was because I wasn't using `add_filter` and I wasn't returning the `$features` array.

File: _tests/tests/__snapshots__/E2eTest__test_e2e_no_op__1.php
Comment (MrJnrman) [reply _tests/tests/__snapshots__/E2eTest__test_e2e_no_op__1.php]: As per DM chat, it turned out that the the `ENABLE_NEW_PRODUCT_EDITOR` where affecting the e2e tests. Changing the name to `QIT_ENABLE_NEW_PRODUCT_EDITOR` did the trick

**Review by Luc45 — APPROVED** [review]
Thank you for these self tests, Jamel! We can merge this once the Manager counterpart is merged.

**Pull request merged**
